google/cloud/storage/internal/object_requests.h

```diff
@@ -263,28 +263,21 @@ std::ostream& operator<<(std::ostream& os, UpdateObjectRequest const& r);
  * Represents a request to the `Objects: compose` API.
  */
 class ComposeObjectRequest
-    : public GenericObjectRequest<ComposeObjectRequest, EncryptionKey,
-                                  Generation, DestinationPredefinedAcl,
-                                  KmsKeyName, IfGenerationMatch,
-                                  IfMetagenerationMatch, UserProject> {
+    : public GenericObjectRequest<
+          ComposeObjectRequest, EncryptionKey, Generation,
+          DestinationPredefinedAcl, KmsKeyName, IfGenerationMatch,
+          IfMetagenerationMatch, UserProject, WithObjectMetadata> {
  public:
   ComposeObjectRequest() = default;
-  explicit ComposeObjectRequest(
-      std::string bucket_name,
-      std::vector<ComposeSourceObject> const& source_objects,
-      std::string destination_object_name,
-      ObjectMetadata destination_object_metadata);
+  explicit ComposeObjectRequest(std::string bucket_name,
+                                std::vector<ComposeSourceObject> source_objects,
+                                std::string destination_object_name);
 
   /// Returns the request as the JSON API payload.
-  std::string json_payload() const { return json_payload_; }
-
-  ObjectMetadata const& destination_metadata() const {
-    return destination_metadata_;
-  }
+  std::string JsonPayload() const;
```

Comment:
Ah, is this a style guide inconsistency?

Reply:
Thanks! I need to be more careful next time 😄

Reply:
Oh, you were doing the right thing.  See the C++ style guide recommends (and some say require) `snake_case` for simple accessors, but it requires `PascalCase` for more complex functions.  So when the implementation changes the naming convention changes.  That is a terrible terrible idea in the style guide.  It should be burnt with fire.

Reply:
ahhhh. oh wow. that's a loaded style guide requirement. I guess that makes sense, but at the same time confusing. Thanks for clarifying.